app/cdap/components/RecipeManagement/ApplyRecipeView.tsx

```diff
@@ -0,0 +1,219 @@
+    setIsOpen(false);
+    setSelectedRecipe(null);
+    const errorMessage = err.message;
+    setSnackbar({
+      open: true,
+      isSuccess: false,
+      message: T.translate(`${PREFIX}.applyRecipeFailed`, {
+        errorDetails: errorMessage,
+      }).toString(),
+    });
+  };
+
+  const closeSnackBar = () => {
+    setSnackbar(() => ({
+      open: false,
+    }));
+  };
+
+  useEffect(() => {
+    if (timeoutId) {
+      clearTimeout(timeoutId);
+    }
+    if (snackbarState.open && snackbarState.isSuccess) {
+      timeoutId = setTimeout(closeSnackBar, 5000);
+    }
+  }, [snackbarState.open]);
+
+  const renderRecipeList = () => {
+    return (
+      <>
+        <h6>{T.translate(`${PREFIX}.selectRecipeToApply`)}</h6>
+        <RecipeList
+          isOpen={isOpen}
+          showAllColumns={false}
+          showActions={false}
+          selectHandler={handleSelectRecipe}
+          sortBy={SortBy.UPDATED}
+          sortOrder={SortOrder.DESCENDING}
+          pageSize={6}
+          showPagination={true}
+          enableSorting={true}
+        />
+      </>
+    );
+  };
+
+  /* TO DO : Remove this function once Recipe Details UI component is integrated  */
+  const renderRecipeDetailsView = () => {
+    const Container = styled.div`
+      display: flex;
+    `;
+    const MainContainer = styled.div`
+      line-height: 2.5;
+    `;
+    const StepsTable = styled.table`
+      box-shadow: 0px 1px 1px rgba(0, 0, 0, 0.2), 0px 2px 2px rgba(0, 0, 0, 0.14),
+        0px 1px 5px rgba(0, 0, 0, 0.12);
+      width: 100%;
+    `;
+    const Divider = styled.div`
+      width: 1px;
+      height: 20px;
+      background-color: #dadce0;
+      margin: 7px 15px;
+    `;
+    return (
+      <>
+        <MainContainer>
+          <div>
+            <strong>{selectedRecipe.recipeName}</strong>
+          </div>
+          <Container>
+            <span>
+              {`${selectedRecipe.recipeStepsCount} `} {T.translate(`${PREFIX}.recipeSteps`)}
+            </span>
+            <Divider></Divider>
+            <span>{format(selectedRecipe.updatedTimeMillis, TYPES.TIMESTAMP_MILLIS)}</span>
+          </Container>
+          <div>{selectedRecipe.description}</div>
+          <div className="table">
+            <StepsTable>
+              <tbody className="table table-hover">
+                <tr>
+                  <th>{'#'}</th>
+                  <th>{T.translate(`${PREFIX}.recipeSteps`)}</th>
+                </tr>
+                {Object.keys(selectedRecipe.directives).map((key, index) => (
+                  <tr className="table-row" key={`tablevalue-${index}`}>
+                    <td>{index}</td>
+                    <td>{selectedRecipe.directives[key] + ' ' + key}</td>
+                  </tr>
+                ))}
+              </tbody>
+            </StepsTable>
+          </div>
+        </MainContainer>
+      </>
+    );
+  };
+
+  const renderRecipeDetailsWithApply = () => {
+    const ButtonsContainer = styled.div`
```

Comment:
If this is not a temo component, move it to the top of the file (after imports), otherwise, ignore this comment.

Reply:
Done